lib/xema/castable.ex

```diff
@@ -302,8 +302,40 @@ defimpl Xema.Castable, for: Map do
     end)
   end
 
+  def cast(map, :list, _module, _schema) do
+    sorted_keys = map |> Map.keys() |> sort_keys()
+
+    {:ok, Enum.map(sorted_keys, fn k -> Map.get(map, k) end)}
```

Comment:
One last point:
```suggestion
    with :error <- to_sorted_list(map, keys: false) do
      {:error, %{to: :map, value: map}}
    end
```
That needs also 
```elixir
defimpl Xema.Castable, for: Map do
  use Xema.Castable.Helper

  import Xema.Utils, only: [to_sorted_list: 2]
  ...
```
some lines above.

And, then the test `cast/2 with list of maps (Xema) sorts the list based on key mixed values` needs to be updated.

And then everything is fine.

Reply:
Thanks. Done in de75041d34791d84bf0f676ce07f57821bdcb1c1